lib/samples/download_preplanned_map_area/download_preplanned_map_area_sample.dart

```diff
@@ -34,130 +32,275 @@ class DownloadPreplannedMapAreaSample extends StatefulWidget {
-                setState(() => _isLoading = false);
-              }
+  void onMapViewReady() async {
+    // Configure the directory to download offline maps to.
+    _downloadDirectory = await createDownloadDirectory();
+
+    // Create a map using a webmap from a portal item.
+    final portal = Portal.arcGISOnline();
+    final portalItem = PortalItem.withPortalAndItemId(
+        portal: portal, itemId: 'acc027394bc84c2fb04d1ed317aac674');
+    _webMap = ArcGISMap.withItem(portalItem);
+
+    // Create and load an offline map task using the portal item.
+    _offlineMapTask = OfflineMapTask.withPortalItem(portalItem);
+    await _offlineMapTask.load();
+
+    // Get the preplanned map areas from the offline map task and load each.
+    final preplannedMapAreas = await _offlineMapTask.getPreplannedMapAreas();
+    for (final mapArea in preplannedMapAreas) {
+      await mapArea.load();
+      // Add each map area as a key in the Map, setting the Job to null initially.
+      _preplannedMapAreas[mapArea] = null;
+    }
+
+    // Initially set the web map to the map view controller.
+    _mapViewController.arcGISMap = _webMap;
+
+    // Upate the state with the preplanned map areas and set the UI state to ready.
+    setState(() => _ready = true);
+  }
+
+  // Builds a map selection widget with a list of available maps.
+  Widget buildMapSelectionSheet(BuildContext context) {
+    return Container(
+      padding: EdgeInsets.fromLTRB(
+          20.0,
+          20.0,
+          20.0,
+          View.of(context).viewPadding.bottom /
+              View.of(context).devicePixelRatio),
+      child: Column(
+        mainAxisSize: MainAxisSize.min,
+        children: [
+          Row(
+            children: [
+              Text('Select Map', style: Theme.of(context).textTheme.titleLarge),
+              const Spacer(),
+              IconButton(
+                icon: const Icon(Icons.close),
+                onPressed: () => setState(() => _mapSelectionVisible = false),
+              ),
+            ],
+          ),
+          const SizedBox(height: 10),
+          // Create a list tile for the web map.
+          ListTile(
+            title: const Text('Web Map (online)'),
+            trailing: _mapViewController.arcGISMap == _webMap
+                ? const Icon(Icons.check)
+                : null,
+            onTap: () => _mapViewController.arcGISMap == _webMap
+                ? setMapAndViewpoint(_webMap)
+                : null,
+          ),
+          const SizedBox(height: 20),
+          Text('Preplanned Map Areas:',
+              style: Theme.of(context).textTheme.titleMedium),
+          const SizedBox(height: 20),
+          ListView.builder(
+            shrinkWrap: true,
+            itemCount: _preplannedMapAreas.length,
+            itemBuilder: (context, index) {
+              return Padding(
+                padding:
+                    const EdgeInsets.symmetric(vertical: 5.0, horizontal: 0),
```

Comment:
For consistency, I've been using '.0' for all parameters that expect doubles. Including SizedBox parameters. But I dunno, that's not really a coding standard yet.